http://bugzilla.gnome.org/show_bug.cgi?id=577799

--- origsrc/gvfs-1.2.1/daemon/Makefile.am	2009-03-05 06:04:27.000000000 -0600
+++ src/gvfs-1.2.1/daemon/Makefile.am	2009-04-02 20:22:11.728201700 -0500
@@ -272,7 +272,7 @@ gvfsd_trash_CPPFLAGS = \
 	-DBACKEND_TYPES='"trash", G_VFS_TYPE_BACKEND_TRASH,' \
 	-Itrashlib
 
-gvfsd_trash_LDADD = $(libraries) trashlib/libtrash.a
+gvfsd_trash_LDADD = trashlib/libtrash.a $(libraries)
 
 gvfsd_computer_SOURCES = \
 	gvfsbackendcomputer.c gvfsbackendcomputer.h \
